src/plugin_discovery/plugin_pack/create_pack.js

```diff
@@ -0,0 +1,30 @@
+import { PluginPack } from './plugin_pack';
+import { createInvalidPackError } from '../errors';
+
+function createPack(packageJson) {
+  let provider = require(packageJson.directoryPath);
+  if (provider.__esModule) {
+    provider = provider.default;
+  }
+  if (typeof provider !== 'function') {
+    throw createInvalidPackError(packageJson.directoryPath, 'must export a function');
+  }
+
+  return new PluginPack({ path: packageJson.directoryPath, pkg: packageJson.contents, provider });
+}
+
+export const createPack$ = (packageJson$) => (
+  packageJson$
+    .mergeMap(({ error, packageJson }) => {
```

Comment:
`.map()` should be sufficient here, and I suggest making this a bit more resilient by either checking specifically for `result.packageJson` and either forwarding any result that doesn't have a `packageJson` property or throwing an informative error when an unexpected result is received.